lib/parsec/fileparse.py

```diff
@@ -71,6 +71,9 @@
     ''',
     re.VERBOSE)
 
+_BAD_CONTINUATION_TRAILING_WHITESPACE = re.compile(
+    r'''.*\\\s''')
```

Comment:
Isn't it matching `\` anywhere? Including comments? Looks like comments are not eliminated before using this regex.

The following suite passes validation in the master branch:

```
# test.rc
[scheduling]
    initial cycle point = 20000101T06
    final cycle point = 20010101T18
    [[dependencies]]
        [[[ T00 ]]]
            # Remember you need to use \  some times!
            graph = """
                foo | bar  => baz & qux
                pub
            """
        [[[ T12 ]]]
            graph = """
                qux
                baz
            """
```

```bash
kinow@ranma:/tmp$ cylc validate test.rc 
Valid for cylc-7.7.1-303-g136dc
```

But fails to validate with this branch

```bash
kinow@ranma:/tmp$ cylc validate test.rc 
FileParseError:
Syntax error line 5: Whitespace after the line break character (\).
```

(Note: removed the other `\` to avoid confusion in this example).

So I think it fixes the issue, but I still don't have enough experience with Cylc and writing suites, to say with confidence if that's OK or not.

@matthewrmshin, could it be a problem? I think besides comments, we could find `\` in other places too? Not sure how strict we are need to be. Other than that, minor nit-pick is only that the other two regexes above the new one had the verbose mode turned on, with some nice comments. But this one is so simple that I think we don't care about that.

Ready to approve if finding `\`'s in other parts of the code is not a problem.

Reply:
For what's worth, here's the cylc version in the second example: 

```bash
kinow@ranma:/tmp$ cylc version
7.7.1-290-gfc51f
```

Which matches the last comment (`g` for git, plus `fc51f6930be956619001d641c1d269117b324fc1` shortened).

Reply:
How about `/^([^#\n]+)?\\\s+$/gm`?

Reply:
That looks like a sensible solution. I shall do that.

Reply:
(I think this is resolved)